argilla/docs/getting_started/huggingface-spaces.md

```diff
@@ -0,0 +1,147 @@
+    Make sure to enable the Hugging Face authentication before adding any data, as the Space will need to be restarted with a "Factory build", removing all the previous data.
+
+1. [Create an OAuth App in Hugging Face](https://huggingface.co/settings/applications/new) and fill in the information:
+
+    | Field        | Value |
+    |--------------|----------|
+    | Homepage URL | [`[Your Argilla Space direct URL]`](#deploy-argilla-on-spaces) |
+    | Logo URL | `[Your Argilla Space direct URL]/favicon.ico` |
+    | Scopes | `openid` and `profile` |
+    | Redirect URL | `[Your Argilla Space Direct URL]/oauth/huggingface/callback`
+
+2. Add the created Client ID and App Secret to your Space. It can be done in two different ways.
+
+    === "As secrets"
+
+        Add them to your Space in "Settings" > "New secret".
+
+        | Secret name         | Value |
+          |-------------------------------|----------|
+          | `OAUTH2_HUGGINGFACE_CLIENT_ID` | [Your Client ID] |
+          | `OAUTH2_HUGGINGFACE_CLIENT_SECRET` | [Your App Secret] |
+
+    === "As environment variables in `.oauth.yaml`"
+
+        Add them to your Space in "Files" > `.oauth.yaml`.
+
+        !!! warning
+            Be aware that the `.oauth.yaml` file is public in public spaces or may be accessible by other members of your organization in private spaces.
+
+            Therefore, we recommend setting these variables as environment secrets.
+
+        ??? quote "Example code in `.oauth.yaml`"
+            ```yaml
+            # This attribute will enable or disable the Hugging Face authentication
+            enabled: true
+
+            providers:
+            # The OAuth provider setup
+            # For now, only Hugging Face is supported
+              - name: huggingface
+                # This is the client ID of the OAuth app. You can find it in your Hugging Face settings.
+                # see https://huggingface.co/docs/hub/oauth#creating-an-oauth-app for more info.
+                # You can also provide it by using the env variable `OAUTH2_HUGGINGFACE_CLIENT_ID`
+                client_id: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXXX
+
+                # This is the client secret of the OAuth app. You can find it in your Hugging Face settings.
+                # See https://huggingface.co/docs/hub/oauth#creating-an-oauth-app for more info.
+                # We encourage you to provide it by using the env variable `OAUTH2_HUGGINGFACE_CLIENT_SECRET`
+                client_secret: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXXX
+
+                # The scope of the OAuth app. At least `openid` and `profile` are required.
+                scope: openid profile
+
+            # This section defines the allowed workspaces for the oauth users.
+            # Workspaces defined here must exist in Argilla.
+            allowed_workspaces:
+                - name: admin
+            ```
+
+3. Check that the `enabled` parameter is set to `true` in "Files" > `.oauth.yaml`.
```

Comment:
Perhaps another thing to add here is that in the `.oauth.yaml` you can change / add allowed workspaces. 
Also somewhere mention that users logging in with oauth will all have the role annotator. If they want to create users with admin or owner roles, they need to be created the standard way and log in with username and password.